src/robusta/core/playbooks/playbooks_event_handler_impl.py

```diff
@@ -182,8 +190,8 @@ def __get_fired_trigger(
         return None
 
     def __handle_findings(self, execution_event: ExecutionBaseEvent):
-        for finding in execution_event.findings:
-            for sink_name in execution_event.named_sinks:
+        for sink_name in execution_event.named_sinks:
+            for finding in execution_event.sink_findings[sink_name]:
```

Comment:
I would update the comment a few lines below about deep copy. I think we do need a deep copy because enrichments are shared between diff. sinks but I would emphasize in the comment that the problem is the enrichments, not the finding itself which is no longer shared.